sw/device/lib/dif/meson.build

```diff
@@ -0,0 +1,15 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# UART DIF library (dif_uart)
+dif_uart = declare_dependency(
+  sources: [hw_ip_uart_reg_h],
```

Comment:
It would be preferred if DIFs avoid exporting the underlying register headers.

Reply:
That's a very good point.

Reply:
Do you mean having *_regs.h only accessible within .../lib/dif?

Reply:
Yes. It would be good to encapsulate the hardware spec inside the DIF, and just export the DIF API.

Reply:
It seems reasonable, the only slight issue I see is that then for higher level drivers/DV that wan't to implement extra functionality - we would need to use a separate set of register definitions. Is that a prefered way, seems like duplicating/maintaining several copies of same register definitions.

Reply:
As per previous discussion, the consensus is that register definitions generated by regtool should not be exposed outside of the DIFs.

Reply:
Meson does not provide a good way to enforce this, unfortunately. Its mechanism for setting up include search paths has proven hard to control... so I would not expect to be able to enforce this.